### PR TITLE
Feature/#8 - Generalized logging support

### PR DESCRIPTION
**Pull request by els0r**
This change allows goProbe to plug in any logger that satisfies the logging interface from https://github.com/els0r/log. Many open source implementations provide the functions demanded by the interface, hence, changing the logger for goProbe can now be done with minimal effort.

The change also makes a backwards compatible alteration to the configuration file, which now allows to specify the logging destination and level explicitly. At this stage it relies on the available loggers from els0r/log. I recommend merging this only upon completion of the Syslog Logger in that package in order to be fully compatible with goProbe's old behavior.

### CODE REVIEW

File: cmd/goQuery/goQuery.go [hunk 1]
```diff
@@ -314,10 +315,12 @@ func throwMsg(msg string, external bool, fmtSpec string) {
 	if fmtSpec == "json" {
 		// If called non-interactively, write full error message to message log
 		if external {
-			if goDB.InitDBLog() != nil {
+			// TODO: change this to syslog
```
Comment (fako1024) [inline cmd/goQuery/goQuery.go]: Is this left undone on purpose (i.e. TBD later) or was it just forgotten during implementation testing?

File: pkg/goDB/DBWorkManager.go [hunk 2]
```diff
@@ -42,15 +42,18 @@ type DBWorkManager struct {
...
 	workloads          []DBWorkload
 	numProcessingUnits int
+
+	logger log.Logger
 }
 
 func NewDBWorkManager(dbpath string, iface string, numProcessingUnits int) (*DBWorkManager, error) {
 	// whenever a new workload is created the logging facility is set up
-	if err := InitDBLog(); err != nil {
+	l, err := log.NewConsoleLogger()
```
Comment (fako1024) [inline pkg/goDB/DBWorkManager.go]: Kept to console on purpose here as well?

Comment (els0r) [reply pkg/goDB/DBWorkManager.go]: Turns out I didn't push the most recent changes... All fixed in recent commit.

**Comment by els0r** [comment]
@fako1024 : ping

File: pkg/goDB/DBWorkManager.go (hunk 2 above)
Comment (fako1024) [reply pkg/goDB/DBWorkManager.go]: Yeah, totally makes sense - I just saw the change from the "old way" and was curious!

**Pull request merged**